core/build.gradle

```
@@ -99,41 +91,25 @@ dependencies {
         exclude group: "*", module: "*"
     }
 
+    compileOnly 'org.jetbrains:annotations:21.0.1'
+
     api 'junit:junit:4.12'
     api 'org.slf4j:slf4j-api:1.7.30'
-    compileOnly 'org.jetbrains:annotations:21.0.1'
-    testCompileClasspath 'org.jetbrains:annotations:21.0.1'
     api 'org.apache.commons:commons-compress:1.20'
-    api ('org.rnorth.duct-tape:duct-tape:1.0.8') {
-        exclude(group: 'org.jetbrains', module: 'annotations')
-    }
-
-    api "com.github.docker-java:docker-java-api:3.2.11"
-
-    shaded ('com.github.docker-java:docker-java-core:3.2.11') {
-        exclude(group: 'com.github.docker-java', module: 'docker-java-api')
-        exclude(group: 'com.github.docker-java', module: 'docker-java-transport')
-        exclude(group: 'com.fasterxml.jackson.core', module: 'jackson-annotations')
-        exclude(group: 'com.google.code.findbug')
-        exclude(group: 'org.slf4j')
-        exclude(group: 'org.apache.commons', module: 'commons-compress')
-    }
-
-    shaded ('com.github.docker-java:docker-java-transport-okhttp:3.2.11') {
-        exclude(group: 'com.github.docker-java', module: 'docker-java-core')
-        exclude(group: 'net.java.dev.jna')
-        exclude(group: 'org.slf4j')
-    }
-
+    api 'org.rnorth.duct-tape:duct-tape:1.0.8'
+    api 'com.github.docker-java:docker-java-api:3.2.11'
     api 'com.github.docker-java:docker-java-transport-zerodep:3.2.11'
 
+    shaded 'com.github.docker-java:docker-java-core:3.2.11'
+    shaded 'com.github.docker-java:docker-java-transport-okhttp:3.2.11', {
+        // TODO remove unused `net.java.dev.jna:jna-platform` from `docker-java-transport-okhttp`
+        exclude(group: 'net.java.dev.jna', module: 'jna-platform')
+    }
     shaded "org.yaml:snakeyaml:1.29"
-
     shaded 'org.glassfish.main.external:trilead-ssh2-repackaged:4.1.2'
+    shaded 'org.zeroturnaround:zt-exec:1.12'
 
-    shaded 'org.zeroturnaround:zt-exec:1.12', {
-        exclude(group: 'org.slf4j')
-    }
+    testCompileClasspath 'org.jetbrains:annotations:21.0.1'
```

```suggestion
    testCompileOnly 'org.jetbrains:annotations:21.0.1'
```

shouldn't be that with `Only` suffix? As *Classpath configurations are intended to be resolvable, but not consumable

---

It was `testCompileClasspath` before, so I suggest we address it separately, as this PR is something massive already :)

---

okay, didn't notice it was moved

---

I will try to polish this PR and reduce the changeset while fixing the merge conflict :+1:

---

Done